code/datums/status_effects/xeno_buffs.dm

```diff
@@ -871,3 +871,499 @@
 	xeno_owner.remove_filter("frenzy_screech_outline")
 	return ..()
+
+// ***************************************
+// *********** Upgrade Chambers Buffs - Survival
+// ***************************************
+/atom/movable/screen/alert/status_effect/upgrade_carapace
+	name = "Carapace"
+	desc = "Armor increased."
+	icon_state = "xenobuff_carapace"
+
+/datum/status_effect/upgrade_carapace
+	id = "upgrade_carapace"
+	duration = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/upgrade_carapace
+	/// Owner typed as an xenomorph.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// The amount of soft armor given.
+	var/armor_buff_per_chamber = 2.5
+	/// The amount of times to multiply the armor buff by.
+	var/chamber_scaling = 0
+
+/datum/status_effect/upgrade_carapace/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	buff_owner = owner
+	chamber_scaling = length(buff_owner.hive.shell_chambers)
+	RegisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_SURVIVAL, PROC_REF(update_buff))
+	buff_owner.soft_armor = buff_owner.soft_armor.modifyAllRatings(armor_buff_per_chamber * chamber_scaling)
+	return TRUE
+
+/datum/status_effect/upgrade_carapace/on_remove()
+	UnregisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_SURVIVAL)
+	buff_owner.soft_armor = buff_owner.soft_armor.modifyAllRatings(-armor_buff_per_chamber * chamber_scaling)
+	return ..()
+
+/// Sets the chamber_scaling to the amount of active survival chambers and adjusts soft armor accordingly.
+/datum/status_effect/upgrade_carapace/proc/update_buff()
+	SIGNAL_HANDLER
+	buff_owner.soft_armor = buff_owner.soft_armor.modifyAllRatings(armor_buff_per_chamber * (length(buff_owner.hive.shell_chambers) - chamber_scaling))
+	chamber_scaling = length(buff_owner.hive.shell_chambers)
+
+// ***************************************
+// ***************************************
+// ***************************************
+/atom/movable/screen/alert/status_effect/upgrade_regeneration
+	name = "Regeneration"
+	desc = "Regeneration increased."
+	icon_state = "xenobuff_regeneration"
+
+/datum/status_effect/upgrade_regeneration
+	id = "upgrade_regeneration"
+	duration = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/upgrade_regeneration
+	/// Owner typed as an xenomorph.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// The amount of max health to be regenerated everytime the proc 'heal_wounds' is called.
+	var/health_regen_per_chamber = 0.015 // 1.5%
+	/// The amount of sunder to be regenerated everytime the proc 'heal_wounds' is called.
+	var/sunder_regen_per_chamber = 0.166
+	/// The amount of times to multiply health/sunder regen by.
+	var/chamber_scaling = 0
+
+/datum/status_effect/upgrade_regeneration/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	buff_owner = owner
+	chamber_scaling = length(buff_owner.hive.shell_chambers)
+	RegisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_SURVIVAL, PROC_REF(update_buff))
```

Comment:
Actually, you could make a shared chamber status type and just call parent to do the chamber stuff there, if you want